cfpb/consumer-credit-trends
# Automate line charts

Adds all the line charts for the auto loans and credit cards markets to the charts.js config; includes compiled static html for those 2 markets' line graphs.

This is a good set of compiled svgs for validating the data and accuracy in the d3 charts. See em here: https://cfarm.github.io/consumer-credit-trends/charts/

This sets up the basic workflow needed to generate all the charts for all markets. Next steps needed for MVP are: 
1. generate the bar charts with these templates (needs JS work)
1. generate maps within templates (needs JS work)
1. add config data to to charts.js for the Mortgage + Student loans line charts, and release to github for review
1. update styles

## Additions

- placeholder for the bar and map charts for the auto loan market. this needs updating to get the d3 code to render properly in the handlebars templates.
- adds CSS for the colors for bar + line graphs for each report type


## Compiled line charts for the auto loan market:
### [All the line charts for auto loans & credit cards on one page!](https://cfarm.github.io/consumer-credit-trends/charts/)

## Changes

- reorganized the auto-loan-charts.js so that:
 - the graph types are handled in separate files, `bar.js`, `line.js` and `tile-map.js`
 - bar charts code was updated to take parameters from the charts.js properties, and to loop through objects in charts.js to render the graphs.
 - origination activity bar graphs (lines 22-48 of charts.js) are the only working examples of bar / map. the rest of the bar charts need updates to bar.js to work with the charts.js setup.

## Testing

- check out branch
- run build steps: setup.sh, gulp build, gulp watch

## Review

- @marteki 
- @mistergone 

[Preview this PR without the whitespace changes](?w=0)

## Screenshots


## Notes

-

## Todos

-

## Checklist
